server/src/main/java/com/cloud/storage/snapshot/SnapshotManager.java

```diff
@@ -34,27 +34,84 @@
-    static final ConfigKey<Integer> SnapshotDailyMax = new ConfigKey<Integer>(Integer.class, "snapshot.max.daily", "Snapshots", "8",
-            "Maximum recurring daily snapshots to be retained for a volume. If the limit is reached, snapshots from the start of the day are deleted so that newer ones can be saved. This limit does not apply to manual snapshots. If set to 0, recurring daily snapshots can not be scheduled.", false, ConfigKey.Scope.Global, null);
-    static final ConfigKey<Integer> SnapshotWeeklyMax = new ConfigKey<Integer>(Integer.class, "snapshot.max.weekly", "Snapshots", "8",
-            "Maximum recurring weekly snapshots to be retained for a volume. If the limit is reached, snapshots from the beginning of the week are deleted so that newer ones can be saved. This limit does not apply to manual snapshots. If set to 0, recurring weekly snapshots can not be scheduled.", false, ConfigKey.Scope.Global, null);
-    static final ConfigKey<Integer> SnapshotMonthlyMax = new ConfigKey<Integer>(Integer.class, "snapshot.max.monthly", "Snapshots", "8",
-            "Maximum recurring monthly snapshots to be retained for a volume. If the limit is reached, snapshots from the beginning of the month are deleted so that newer ones can be saved. This limit does not apply to manual snapshots. If set to 0, recurring monthly snapshots can not be scheduled.", false, ConfigKey.Scope.Global, null);
-    static final ConfigKey<Boolean> usageSnapshotSelection = new ConfigKey<Boolean>("Usage", Boolean.class, "usage.snapshot.virtualsize.select", "false",
-            "Set the value to true if snapshot usage need to consider virtual size, else physical size is considered ", false);
-    public static final ConfigKey<Integer> BackupRetryAttempts = new ConfigKey<Integer>(Integer.class, "backup.retry", "Advanced", "3",
-            "Number of times to retry in creating backup of snapshot on secondary", false, ConfigKey.Scope.Global, null);
-
-    public static final ConfigKey<Integer> BackupRetryInterval = new ConfigKey<Integer>(Integer.class, "backup.retry.interval", "Advanced", "300",
-            "Time in seconds between retries in backing up snapshot to secondary", false, ConfigKey.Scope.Global, null);
+    Integer defaultDiskSnapshotMaxHourly = 8;
+    Integer defaultDiskSnapshotMaxDaily = 8;
+    Integer defaultDiskSnapshotMaxWeekly = 8;
+    Integer defaultDiskSnapshotMaxMonthly = 12;
+    Boolean defaultDiskSnapshotUsageVirtualsizeSelect = false;
+    Integer defaultBackupRetry = 3;
+    Integer defaultBackupRetryInterval = 300;
+    Integer snapshotDeltaMax = 16;
+
+    ConfigKey<Integer> snapshotHourlyMax = new ConfigKey<Integer>(Integer.class,
```

Comment:
why did you remove the static and  final keywords here? 

It is rare to see a proper use of them in ACS, and here it was being used in the right manner.

Reply:
@rafaelweingartner Can you explain why the keywords are needed here? I would like to understand your comment